### ISSUE TITLE
Certificate based RSA public key creation.

### ISSUE THREAD

**Issue by hhanesand** [vapor/jwt#46]
Would be nice to have a way to create a key from a certificate containing a public key. Its not easily doable as of now because the initializes of the `RSxxx` classes require `Bytes` input. The following code is possible with #45

```swift
extension RSASigner {
    
    init(certificate: String) throws {
        let certificateBytes = certificate.bytes
        
        let cert_bio = BIO_new_mem_buf(certificateBytes, Int32(certificateBytes.count))
        var _cert: UnsafeMutablePointer<X509>?
        PEM_read_bio_X509(cert_bio, &_cert, nil, nil)
        
        guard let cert = _cert else {
            throw Abort.custom(status: .internalServerError, message: "Failed to decode certificate.")
        }
        
        guard let publicKey = X509_get_pubkey(cert) else {
            throw Abort.custom(status: .internalServerError, message: "Failed to decode public key.")
        }
        
        guard let rsa = EVP_PKEY_get1_RSA(publicKey) else {
            throw Abort.custom(status: .internalServerError, message: "Failed to get rsa key.")
        }
        
        self.init(key: .public(rsa))
    }
}
```

**Comment by vzsg** [comment]
The topic of making the C classes public came up previously, and I voted for *not* exposing them, because:
1. The fact that the library uses Libre/OpenSSL is an implementation detail, which might change in the future. Having a clean, Swift-only public interface helps the future maintainers and users keep their sanity.
1. The low-level C interface is a risky business. In your example code above, you've already leaked the BIO in `cert_bio` and the X509 certificate in `_cert` as you forgot to call `BIO_free` and `X509_free`. (Those should preferably be in a `defer` statement).

**tl;dr**: I think #45 is a bad idea. However, initializing keys from PEM files would be a nice addition to the library, as long as it's _inside_ the library and doesn't leak memory.

**Comment by hhanesand** [comment]
Hmm, true. As you point out, its very hard for someone (like me) with no C background to properly code in C.
